SwissTransferFeatures/UploadProgressView/TransferSessionManager.swift

```diff
@@ -0,0 +1,159 @@
+/*
+ Infomaniak SwissTransfer - iOS App
+ Copyright (C) 2024 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Combine
+import Foundation
+import InfomaniakCore
+import InfomaniakDI
+import OSLog
+import STCore
+import STNetwork
+import SwissTransferCore
+
+final class UploadTaskDelegate: NSObject, URLSessionTaskDelegate {
+    let taskProgress: Progress
+
+    init(totalBytesExpectedToSend: Int) {
+        taskProgress = Progress(totalUnitCount: Int64(totalBytesExpectedToSend))
+    }
+
+    func urlSession(
+        _ session: URLSession,
+        task: URLSessionTask,
+        didSendBodyData bytesSent: Int64,
+        totalBytesSent: Int64,
+        totalBytesExpectedToSend: Int64
+    ) {
+        taskProgress.completedUnitCount = totalBytesSent
+    }
+}
+
+class TransferSessionManager: ObservableObject {
+    @LazyInjectService private var injection: SwissTransferInjection
+
+    @Published var percentCompleted: Double = 0
+
+    private let uploadUrlSession = URLSession.shared
```

Comment:
Should it be URL instead of Url?
```suggestion
    private let uploadURLSession = URLSession.shared
```